src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -0,0 +1,48 @@
+import { ArrowForwardIcon } from '@chakra-ui/icons'
+import { Button, HStack, Stack } from '@chakra-ui/react'
+import { CAIP19 } from '@shapeshiftoss/caip'
+import { NavLink } from 'react-router-dom'
+import { Card } from 'components/Card/Card'
+import { Text } from 'components/Text'
+import { useRenderForChains } from 'hooks/useRenderForChains/useRenderForChains'
+import { AccountSpecifier } from 'state/slices/portfolioSlice/portfolioSlice'
+
+import { StakingOpportunitiesRow } from './StakingOpportunitiesRow'
+
+type StakingOpportunitiesProps = {
+  tokenId?: string
+  assetId: CAIP19
+  accountId?: AccountSpecifier
+  isLoaded?: boolean
+}
+
+export const StakingOpportunities = ({ assetId: caip19 }: StakingOpportunitiesProps) => {
+  // TODO: wire up with real validator data
+  const validators = [{ name: 'Cosmos Validator' }]
+
+  const cosmosStaking = (
+    <Card>
+      <Card.Header flexDir='row' display='flex'>
+        <HStack justify='space-between' flex={1}>
+          <Card.Heading>
+            <Text translation='staking.staking' />
+          </Card.Heading>
+
+          <Button size='sm' variant='link' colorScheme='blue' as={NavLink} to='/defi/earn'>
+            <Text translation='common.seeAll' /> <ArrowForwardIcon />
+          </Button>
+        </HStack>
+      </Card.Header>
+      <Card.Body pt={0}>
+        <Stack spacing={2} mt={2} mx={-4}>
+          {validators.map(validator => (
+            <StakingOpportunitiesRow name={validator.name} />
+          ))}
+        </Stack>
+      </Card.Body>
+    </Card>
+  )
+
+  // TODO: add ChainTypes.Cosmos to the array to render only on cosmos based chains
+  return useRenderForChains([], () => cosmosStaking)
```

Comment:
Nice use of closures here! 

Speaking of them, and probably nitpick:
Would we even need to pass some arguments to the anonymous function here when calling back `renderFn`, given that the `StakingOpportunities` closure has all we need?

1. If so, `cosmosStaking` and the next ones to come could be made a component and used like so in `useRenderForChains` (that's effectively what it is by doing `() => someJsxNode` here, `() => cosmosStaking` evaluates to `React.FC`)
2. Else, we can ditch the `() =>` part

Reply:
yeah, also i wouldn't prefix this as a hook, just a util, or react will get confused and sad. `renderForChains()` is prob best.

also, you can `useMemo` the component above

Reply:
Good call. That makes sense

Reply:
after discussing because `useRenderForChains` includes a hook itself this is fine

Reply:
@gomesalexandre i like the closure because we can lazily render, and inject props in future. you're right, this component doesn't need any, but i can def foresee components that will.